src/mobu/services/business/notebookrunner.py

```diff
@@ -68,28 +69,33 @@ def annotations(self, cell_id: str | None = None) -> dict[str, str]:
         return result
 
     async def startup(self) -> None:
-        if not self._repo:
-            self.clone_repo()
+        if self._repo_dir is None:
+            self._repo_dir = Path(TemporaryDirectory().name)
+            await self.clone_repo()
         self._notebook_paths = self.find_notebooks()
         self.logger.info("Repository cloned and ready")
         await super().startup()
 
-    def clone_repo(self) -> None:
+    async def shutdown(self) -> None:
+        shutil.rmtree(str(self._repo_dir))
+        self._repo_dir = None
+        await super().shutdown()
+
+    async def clone_repo(self) -> None:
         url = self.options.repo_url
         branch = self.options.repo_branch
-        path = self._repo_dir.name
         with self.timings.start("clone_repo"):
-            self._repo = Repo.clone_from(url, path, branch=branch)
+            await self._git.clone("-b", branch, url, str(self._repo_dir))
 
     def find_notebooks(self) -> list[Path]:
         with self.timings.start("find_notebooks"):
-            notebooks = [
-                p
-                for p in Path(self._repo_dir.name).iterdir()
-                if p.suffix == ".ipynb"
-            ]
+            notebooks: list[Path] = []
+            if self._repo_dir is not None:
+                notebooks = [
+                    p for p in self._repo_dir.iterdir() if p.suffix == ".ipynb"
+                ]
```

Comment:
In any case it's not going to happen, because that git client creation has a specific repo set during creation, but mypy doesn't know that.  I'll make it a ValueError.